### ISSUE TITLE
code return of bin/elasticsearch script should be compatible with init.d deamon

### ISSUE THREAD

**Issue by durandx** [elastic/elasticsearch#13389]
The script under "/bin/elasticsearch" is not compliant with the init.d deamon.

Version tested: 1.5.2
https://github.com/elastic/elasticsearch/blob/1.5/bin/elasticsearch#L162

For example if your JVM crash at startup (for example you want to allocate much HEAP than the JVM can handle with ES_HEAP_SIZE) this script will exit with EXIT_SUCCESS (0).

This will lead to incorrect behaviour of init.d.

For exemple with systemd, the daemon status will look like this : 
```bash
# systemctl status elasticsearch
● elasticsearch.service - LSB: Starts elasticsearch
   Loaded: loaded (/etc/init.d/elasticsearch)
   Active: active (exited) since Tue 2015-09-08 09:46:55 CEST; 7s ago
  Process: 7931 ExecStop=/etc/init.d/elasticsearch stop (code=exited, status=0/SUCCESS)
  Process: 7946 ExecStart=/etc/init.d/elasticsearch start (code=exited, status=0/SUCCESS)
```

Please, Fix the return status of /bin/elasticsearch in case of a JVM crash.

**Comment by clintongormley** [comment]
Given that we have to background the task, any ideas what changes we can make to achieve what you are after?

**Comment by rmuir** [comment]
This is already fixed for 2.0

```
rmuir@beast:~/workspace/elasticsearch/distribution/zip/target/releases/elasticsearch-3.0.0-SNAPSHOT$ ES_HEAP_SIZE=100000 bin/elasticsearch
Error occurred during initialization of VM
Too small initial heap
rmuir@beast:~/workspace/elasticsearch/distribution/zip/target/releases/elasticsearch-3.0.0-SNAPSHOT$ echo $?
1
```
